models/concerns/searchable.rb

```diff
@@ -10,14 +10,6 @@ module Searchable
     after_update :update_indexed_document
     after_destroy :delete_document
 
-    def self.put_search_index_mapping(index=nil)
```

Comment:
We decided on the following:
- in rake, always create the mappings with the index (as we had been).
- create them using separate put_mapping api, exposing a strict ordering requirement when adding the document types
- added a validate_index rake task to ensure the mappings had been added.